numba_dppy/compiler.py

```diff
@@ -58,6 +59,8 @@ class DPPYCompiler(CompilerBase):
     def define_pipelines(self):
         # this maintains the objmode fallback behaviour
         pms = []
+        self.state.parfor_diagnostics = ExtendedParforDiagnostics()
+        self.state.metadata['parfor_diagnostics'] = self.state.parfor_diagnostics
```

Comment:
Is `'parfor_diagnostics'`  attached to the `state.metadata` or `lowerer.metadata` or both? If both can you please explain why?

Reply:
Parfor diagnostics is attached to the state.metadata, and then in SpirvFriendlyLowering it is transferred to DPPYLower
That is, the same diagnostics is used in the lowerer.

Reply:
Makes sense